<a href='https://www.darshan.ac.in/'> <img src='https://www.darshan.ac.in/Content/media/DU_Logo.svg' width="150" height="200"/></a>
<pre>
<center><b><h2> Python for Data Science - 2305CS303 </b></center>
<center><b><h2>Lab - 9</b></center>    
<left><b><h3>Roll No. : 106</b></left>  
<left><b><h3>Name : Ghorecha Harsh</b></left>   
<pre>    

### 1. Create a Pandas Series containing names of 5 students.

In [9]:
import pandas as pd
import numpy as np

name =pd.Series(['Harsh','Darshan','udit','daku','chhota_mumbai'])
print(name)

0            Harsh
1          Darshan
2             udit
3             daku
4    chhota_mumbai
dtype: object


### 2. Create a Series with student roll numbers as index and their IAT scores as values..

In [11]:
s1 = pd.Series(np.random.randint(50,99,5),np.arange(1,6))
s1

1    81
2    59
3    74
4    75
5    54
dtype: int64

### 3. Create a time series (daily) from 2025-08-01 to 2025-08-10 representing attendance tracking for a student.

In [13]:
dates = pd.date_range(start='2025-08-01', end='2025-08-10')
attendance = pd.Series(np.random.choice(['P     ','A'], size = 10),dates)
attendance


2025-08-01    P     
2025-08-02         A
2025-08-03         A
2025-08-04         A
2025-08-05         A
2025-08-06    P     
2025-08-07         A
2025-08-08    P     
2025-08-09    P     
2025-08-10    P     
Freq: D, dtype: object

### 4. Create a DataFrame for 10 students with the following columns: Roll No, Name, PDS, CA, CN, IAT.
#### (Use NumPy random module to generate scores)

In [43]:
rollno = np.arange(101, 111)
name = pd.Series(np.random.choice(['HARSH','DARSHAN','UDIT','KAMO','UDAY','JAY','RONKO','YASH','NIKHIL','BAPUJI'],size = 10))
pds = np.random.randint(60,100, size = 10)
ca = np.random.randint(60,100, size = 10)
cn = np.random.randint(60,100, size = 10)
iat = np.random.randint(60,100, size = 10)

df = pd.DataFrame({'Roll No': rollno, 'Name': name,'PDS': pds,'CA':ca,'CN':cn,'IAT':iat})

df

Roll No     Name  PDS  CA  CN  IAT
0      101      JAY   98  90  62   74
1      102    HARSH   69  89  85   62
2      103    HARSH   95  60  98   89
3      104  DARSHAN   66  78  60   96
4      105   NIKHIL   67  87  77   60
5      106      JAY   83  88  94   94
6      107      JAY   67  79  89   64
7      108   BAPUJI   71  92  84   70
8      109   NIKHIL   97  89  91   72
9      110     KAMO   74  61  72   83

### 5. Display the first 3 rows of the DataFrame.

In [17]:
df.head(3)

Roll No:    Name :  PDS:   CA :  CN :  IAT :
0        101  DARSHAN     82    82    81     83
1        102     UDIT     77    83    74     91
2        103     UDAY     88    96    83     79

### 6. Display the last 2 rows of the DataFrame.

In [19]:
df.tail(2)

Roll No:    Name :  PDS:   CA :  CN :  IAT :
8        109  DARSHAN     72    63    98     85
9        110     UDIT     82    88    95     95

### 7. Use .describe() to summarize the numeric data.

In [21]:
df.describe()

Roll No:       PDS:        CA :       CN :     IAT :
count   10.00000  10.000000  10.000000  10.000000  10.00000
mean   105.50000  84.700000  78.600000  85.900000  84.70000
std      3.02765  11.935475  11.890239  10.795575  10.01166
min    101.00000  62.000000  63.000000  64.000000  68.00000
25%    103.25000  78.250000  69.250000  81.500000  79.75000
50%    105.50000  85.000000  78.500000  87.500000  84.00000
75%    107.75000  94.750000  86.750000  94.750000  92.50000
max    110.00000  98.000000  96.000000  98.000000  99.00000

### 8. Select only the Name column.

In [37]:
df.Name

0       KAMO
1      RONKO
2        JAY
3       KAMO
4    DARSHAN
5     NIKHIL
6     NIKHIL
7      HARSH
8       KAMO
9        JAY
Name: Name, dtype: object

### 9. Select the columns PDS, CN, and IAT.

In [45]:
df[['PDS','CN','IAT']]

PDS  CN  IAT
0   98  62   74
1   69  85   62
2   95  98   89
3   66  60   96
4   67  77   60
5   83  94   94
6   67  89   64
7   71  84   70
8   97  91   72
9   74  72   83

### 10. Select the row with Roll No = 105 using loc.


In [61]:
df.loc[4]

Roll No       105
Name       NIKHIL
PDS            67
CA             87
CN             77
IAT            60
Name: 4, dtype: object

### 11. Select the 4th row using iloc.

In [63]:
df.loc[3]

Roll No        104
Name       DARSHAN
PDS             66
CA              78
CN              60
IAT             96
Name: 3, dtype: object

### 12. Select students with marks in PDS > 80.

In [69]:
 df[df['PDS'] > 80]

Roll No    Name  PDS  CA  CN  IAT
0      101     JAY   98  90  62   74
2      103   HARSH   95  60  98   89
5      106     JAY   83  88  94   94
8      109  NIKHIL   97  89  91   72

### 13. Select students with marks in CA < 70. 

In [73]:
df[df['CA']<70]

Roll No   Name  PDS  CA  CN  IAT
2      103  HARSH   95  60  98   89
9      110   KAMO   74  61  72   83

### 14. Select students with marks in CN > 85 and PDS > 80

In [77]:
df[(df['CN']>85) & (df['PDS']>80)]

Roll No    Name  PDS  CA  CN  IAT
2      103   HARSH   95  60  98   89
5      106     JAY   83  88  94   94
8      109  NIKHIL   97  89  91   72

### 15. Add a new column Total Marks = PDS + CA + CN + IAT.

In [83]:
df['Total Marks'] = df['PDS'] + df['CA'] + df['CN'] + df['IAT']
df

Roll No     Name  PDS  CA  CN  IAT  Total Marks
0      101      JAY   98  90  62   74          324
1      102    HARSH   69  89  85   62          305
2      103    HARSH   95  60  98   89          342
3      104  DARSHAN   66  78  60   96          300
4      105   NIKHIL   67  87  77   60          291
5      106      JAY   83  88  94   94          359
6      107      JAY   67  79  89   64          299
7      108   BAPUJI   71  92  84   70          317
8      109   NIKHIL   97  89  91   72          349
9      110     KAMO   74  61  72   83          290

### 16. Create a new DataFrame of students with Total Marks > 320.

In [93]:
df1= df[df['Total Marks'] > 320]
df1

Roll No    Name  PDS  CA  CN  IAT  Total Marks
0      101     JAY   98  90  62   74          324
2      103   HARSH   95  60  98   89          342
5      106     JAY   83  88  94   94          359
8      109  NIKHIL   97  89  91   72          349